Etities/Refrigerator.cs

```diff
@@ -0,0 +1,208 @@
+                    item.IdOfShelf = shelf.ShelfId;                 
+                    Console.WriteLine("item added successfuly");
+                    return;
+                }
+            }
+            Console.WriteLine("there is no space in the refrigeratot");
+        }
+        public Item RemoveItemToRefrigerator(int id)
+        {
+            foreach (Shelf shelf in Shelves)
+            {
+                foreach(Item item in shelf.Items)
+                {
+                    if (item.ItemId == id)
+                    {
+                        shelf.Items.Remove(item);
+                        item.IdOfShelf = 0;
+                        return item;
+                    }
+                }
+            }
+            Console.WriteLine("item not found");
+            return null;
+
+        }
+        public void cleanRefrigerator()
+        {
+            foreach (Shelf shelf in Shelves)
+            {
+                shelf.Items.RemoveAll(item => item.ExpiryDate <= DateTime.Now);
+            }
+                Console.WriteLine("Refrigerator cleaned successfully.");
+        }
+        public List<Item> WhatIwantToEat(Kosher kosher, TypeOfFood typeOfFood)
+        {
+            List<Item> availbleFoods = new List<Item>();
+            foreach (Shelf shelf in Shelves)
+            {
+                foreach (Item item in shelf.Items)
+                {
+                    if (item.ExpiryDate >= DateTime.Now && item.TypeOfFood == typeOfFood && item.Kosher == kosher)
+                    {
+                       availbleFoods.Add(item);
+                    } 
+                }
+            }
+            return availbleFoods;
+        }
+        public List<Item> SortItemsByExpiryDate()
+        {
+            List<Item> sortedItems = new List<Item>();
+            foreach(Shelf shelf in Shelves)
+                sortedItems.AddRange(shelf.Items);
+            sortedItems.Sort((x, y) => DateTime.Compare(x.ExpiryDate, y.ExpiryDate));
+            return sortedItems;
+        }
+        public List<Shelf> SortShelvesByFreeSpace(List<Shelf> shelves)
+        {
+            List<Shelf> sortedShelves = new List<Shelf>();
+            foreach (Shelf shelf in shelves)
+            {
+                sortedShelves.Add(shelf);
+            }
+            sortedShelves.Sort((x, y) => x.SpaceInTheShelf().CompareTo(y.SpaceInTheShelf()));
+            return sortedShelves;
+        }
+
+        public List<Refrigerator> SorRefrigeratorsByFreeSpace(List<Refrigerator>refrigerators)
+        {
+            List<Refrigerator>freeSpaces = new List<Refrigerator>();
+            foreach (Refrigerator refrigerator in refrigerators)
+            {
+                freeSpaces.Add(refrigerator);
+            }
+            freeSpaces.Sort((x,y)=>x.CalculateRemainingSpaceInRefrigerator().CompareTo(y.CalculateRemainingSpaceInRefrigerator()));
+            return freeSpaces;
+        }
+        public void is20Free()
```

Comment:
this name is not by conventions, and all the logic here should be part of the console app and not of the class and all the prints should hint about that.